### PR TITLE
Annot dec2018

### PR DESCRIPTION
**Pull request by kouichi-c-nakamura**
# What this PR does

1. The builtin class `inputParser`'s deprecated `addParamValue` method has been replaced with `addParameter`.
2. Removed some commas that were causing warning from MATLAB code analyzer
3. Added `updateMapAnnotation` MATLAB function which allows you to change the content of a MapAnnotation object in OMERO server while keeping the ID of the object.

```matlab
function ma = updateMapAnnotation(session, ma, keyvalue, varargin)
% updateMapAnnotation will update the content (key-value pairs) of
% MapAnnotation ma while maintaining its ID.
%
% SYNTAX
% ma = updateMapAnnotation(session, ma, keyvalue)
% ma = updateMapAnnotation(____,'Param',value)
%
% INPUT ARGUMENTS
% session     omero.api.ServiceFactoryPrxHelper object
%
% ma          MapAnnotationI object
%
% keyvalue    cell array of characters | string array
%             The number of columns must be 2. The first colum is for keys
%             and the second column is for values.
%
% OPTIONAL PARA<ETER/VALUE PAIRS
% 'namespace' char
%             Namespace for the MapAnnotation. If you specify this, the
%             value of 'iseditable' will be ignored.
%
% 'description'
%             char
%             Description for the MapAnnotation. 
%
% 'group'     a positive integer
%             group ID
%
% 'iseditable' 
%             false (default) | true | 0 | 1 
%             If true or 1, MapAnnotation (Key-Value Pairs) will
%             be editable via GUI (OMERO.web or OMERO.insight). If you
%             specify 'namespace', 'iseditable' will be ignored.
%
%
% OUTPUT ARGUMENTS
% ma          MapAnnotationI object
%
%
% Written by Kouichi C. Nakamura Ph.D.
% MRC Brain Network Dynamics Unit
% University of Oxford
% kouichi.c.nakamura@gmail.com
% 05-Dec-2018 18:23:13
%
% See also
% writeMapAnnotation, strToMapAnnotation, mapAnnotationToCellstr

```

# Testing this PR

1. required setup: You need an image with MapAnnotation on OMERO server

2. actions to perform. 

```matlab
session = client.createSession(user, password);
ma1 = getObjectAnnotations(session, 'map', 'image', imageID)
s = string(mapAnnotationToCellstr(ma1))
s_ = ["New key","new value";s]
ma2 = updateMapAnnotation(session, ma1, s_, 'iseditable',true)

% check the MapAnnotation form GUI now
% Confirm that it is editable.

ma3 = getObjectAnnotations(session, 'map', 'image', imageID)
s = string(mapAnnotationToCellstr(ma3))
s_ = ["New key 2","new value 2";s]
ma4 = updateMapAnnotation(session, ma3, s_, 'iseditable',false)

% check the MapAnnotation form GUI now
% Confirm that it is editable.
```

You can find `mapAnnotationToCellstr` from [here](https://github.com/openmicroscopy/openmicroscopy/blob/5f23736f11228fce67133592c3bd69541eb0744c/components/tools/OmeroM/src/helper/mapAnnotationToCellstr.m)

3. expected observations

+ The MapAnnotation will be swtiched from editable to uneditable etc.
+ ID of the MapAnnotation will stay the same.
+ The new keys and values will be added to the MapAnnotation.

# Related reading

Link to cards, tickets, other PRs:

https://github.com/openmicroscopy/openmicroscopy/pull/5792

https://github.com/openmicroscopy/openmicroscopy/pull/5799

### CODE REVIEW

File: components/tools/OmeroM/src/annotations/getAnnotations.m [hunk 1]
```diff
@@ -10,9 +10,33 @@
...
+% annotations Java objects of the following classes:
+%             omero.model.CommentAnnotationI | omero.model.DoubleAnnotationI | 
+%             omero.model.FileAnnotationI | omero.model.LongAnnotationI | 
+%             omero.model.MapAnnotationI | omero.model.TagAnnotationI |
+%             omero.model.TimestampAnnotationI | omero.model.XmlAnnotationI
+%
 % See also: GETANNOTATIONTYPES, GETDOUBLEANNOTATIONS, GETCOMMENTANNOTATIONS,
 % GETFILEANNOTATIONS, GETLONGANNOTATIONS, GETTAGANNOTATIONS,
-% GETTIMESTAMPANNOTATIONS, GETXMLANNOTATIONS
+% GETTIMESTAMPANNOTATIONS, GETXMLANNOTATIONS, getObjectAnnotations
```
Comment (jburel) [inline components/tools/OmeroM/src/annotations/getAnnotations.m]: should be GETOBJECTANNOTATIONS to match the rest

File: components/tools/OmeroM/src/annotations/getObjectAnnotations.m [hunk 2]
```diff
@@ -53,8 +56,56 @@
...
+% annotationType     
+%             'comment' | 'double' | 'file' | 'long' | 'map' | 'tag' | ...
+%             'timestamp' | 'xml'
+%             Specifies the type of annotation.
+%             defined by getAnnotationTypes.m
+%
+% parentType     
+%             'project' | 'dataset' | 'image' | 'screen' | 'plate' | ...
+%             'plateacquisition' | 'roi'
+%             Specifies the type of the parent object(s).
```
Comment (jburel) [inline components/tools/OmeroM/src/annotations/getObjectAnnotations.m:72]: extra ``.``

File: components/tools/OmeroM/src/annotations/getObjectAnnotations.m [hunk 3]
```diff
@@ -53,8 +56,56 @@
...
+%             defined by getObjectTypes.m
+%
+%
+% ids         a positive integer | a vector of positive integers | []
+%             ID(s) for the parent object(s)
+%
+% parents     an array of parent Java objects
+%
+% OPTIONAL PARAMETER/VALUE PAIRS
+% 'include'   charcters | cell array of characters
```
Comment (jburel) [inline components/tools/OmeroM/src/annotations/getObjectAnnotations.m]: typo it should be ``characters``

File: components/tools/OmeroM/src/annotations/getObjectAnnotations.m [hunk 4]
```diff
@@ -53,8 +56,56 @@
...
+%             (Optional) Namespace of annotations to be included
+%
+% 'exclude'   charcters | cell array of characters
```
Comment (jburel) [inline components/tools/OmeroM/src/annotations/getObjectAnnotations.m]: same

File: components/tools/OmeroM/src/annotations/getObjectAnnotations.m [hunk 5]
```diff
@@ -53,8 +56,56 @@
...
+% OUTPUT ARGUMENTS
+% anns        Java objects of the following classes:
+%             omero.model.CommentAnnotationI | omero.model.DoubleAnnotationI | 
+%             omero.model.FileAnnotationI | omero.model.LongAnnotationI | 
+%             omero.model.MapAnnotationI | omero.model.TagAnnotationI |
+%             omero.model.TimestampAnnotationI | omero.model.XmlAnnotationI
+%
 % See also: GETIMAGEFILEANNOTATIONS, GETIMAGETAGANNOTATIONS,
-% GETIMAGECOMMENTANNOTATIONS
+% GETIMAGECOMMENTANNOTATIONS, getAnnotations
```
Comment (jburel) [inline components/tools/OmeroM/src/annotations/getObjectAnnotations.m]: GETANNOTATIONS

File: components/tools/OmeroM/src/annotations/getAnnotations.m (hunk 1 above)
Comment (kouichi-c-nakamura) [reply components/tools/OmeroM/src/annotations/getAnnotations.m]: Agreed

File: components/tools/OmeroM/src/annotations/getObjectAnnotations.m (hunk 3 above)
Comment (kouichi-c-nakamura) [reply components/tools/OmeroM/src/annotations/getObjectAnnotations.m]: Thanks

File: components/tools/OmeroM/src/annotations/getObjectAnnotations.m (hunk 5 above)
Comment (kouichi-c-nakamura) [reply components/tools/OmeroM/src/annotations/getObjectAnnotations.m]: Thanks

File: components/tools/OmeroM/src/annotations/getObjectAnnotations.m (hunk 4 above)
Comment (kouichi-c-nakamura) [reply components/tools/OmeroM/src/annotations/getObjectAnnotations.m]: Thanks

**Comment by jburel** [comment]
Commits included in this PR have been migrated to
* https://github.com/ome/omero-matlab/pull/7
* https://github.com/ome/omero-matlab/pull/16
* https://github.com/ome/omero-matlab/pull/17

**Pull request closed without merging**